### ISSUE TITLE
Support native objc Gradle building

### ISSUE THREAD

**Issue by advayDev1** [brunobowden/j2objc-gradle#76]
I don't use Xcode directly**, instead I prefer to use the Gradle wrapper for Xcode clang toolchain.
See https://docs.gradle.org/current/userguide/nativeBinaries.html

All functionality (that I need at least) of both j2objcc and Xcode pods can be done in Gradle directly.  So I'm basically using the j2objcTranslate functionality, and then doing the rest in gradle (without j2objcTest or xcode pods).  My current setup is at:
https://gist.github.com/advayDev1/636f34fa80164645e205
This sets up normal Gradle tasks for building my objc libraries, building test executables, and running tests.

I think folks might want to use Xcode pods or Gradle builds, and the core plugin probably should not enforce either.  Neither actually are very coupled to j2objc, which is purely translation.  I propose we have a tools/ subdirectory or similar that supports pods and gradle native builds, so users can choose what they want.

** For many reasons, most importantly:
- Xcode doesn't handle non-flat source directories very well.
- I prefer to use one build toolchain (and one IDE when needed, IDEA/Android Studio) without denormalization/duplication of configuration.
- Using a single Gradle build ensures a single presubmit/build command builds and tests all platforms and all flavors of my projects.

**Comment by brunobowden** [comment]
This is very interesting Advay. We should discuss this in person when we
meet up.

**Comment by advayDev1** [comment]
@brunobowden Yep will do.

@confile pointed out that perhaps we should be directly using pods on another issue, without the raw copy.  This issue addresses yet another alternative.  Maybe we can continue the discussion here.

**Comment by advayDev1** [comment]
btw, i upgraded one of my projects to gradle 2.4 today and using native
clang compiler is way faster/more parallel (and gets incremental build
correct) than exec-ing j2objcc (for the j2objcCompile task)...

**Comment by brunobowden** [comment]
Can you make this an option? Then we can consider switching it to the
default over time.

**Comment by advayDev1** [comment]
I'll document some stuff for our Tuesday get together and we can decide
whether it is the way to go.  SG?

**Comment by confile** [comment]
@advayDev1 Not sure what you are talking about but I will have a look at your documentation.

**Comment by advayDev1** [comment]
For now take a look at the gist linked to from the first post.  It is close
to what I have running internally and should be self explanatory if you
also read the gradle native build link in the same post.

Re my last comment: In gradle 2.4 they made native compiles run in parallel
according to the number of cores you have.
